data/templates/default.latex

```diff
@@ -152,51 +177,49 @@ $endif$
 $endif$
 % use upquote if available, for straight quotes in verbatim environments
 \IfFileExists{upquote.sty}{\usepackage{upquote}}{}
-% use microtype if available
-\IfFileExists{microtype.sty}{%
-\usepackage[$for(microtypeoptions)$$microtypeoptions$$sep$,$endfor$]{microtype}
-\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
+\IfFileExists{microtype.sty}{% use microtype if available
+  \usepackage[$for(microtypeoptions)$$microtypeoptions$$sep$,$endfor$]{microtype}
+  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
 }{}
 $if(indent)$
 $else$
 \makeatletter
-\@ifundefined{KOMAClassName}{%
+\@ifundefined{KOMAClassName}{% if non-KOMA class
   \IfFileExists{parskip.sty}{%
     \usepackage{parskip}
   }{% else
     \setlength{\parindent}{0pt}
     \setlength{\parskip}{6pt plus 2pt minus 1pt}}
-}{% else
+}{% if KOMA class
   \KOMAoptions{parskip=half}}
 \makeatother
 $endif$
 $if(verbatim-in-note)$
 \usepackage{fancyvrb}
 $endif$
-$if(colorlinks)$
-\usepackage{xcolor}
-$endif$
-\usepackage{hyperref}
+\IfFileExists{xcolor.sty}{\usepackage{xcolor}}{}
```

Comment:
What happens if they specify `colorlinks` but `xcolor` doesn't exist?
Previously we would have gotten a latex compilation failure, pointing to the need for xcolor.
What do we get in this case now?

Reply:
Good point: if there are no visible links in the document, it will work even if `colorlinks` is specified.

It fails when the document tries to render a colour using the `xcolor` definitions: for example, in a document with a URL and `colorlinks` enabled:

```latex
! LaTeX Error: Undefined color `Blue'.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.69 \href{https://pandoc.org}{test url}
```

Is it worse, then, to require `xcolor` even in situations when it isn't needed, or to have an error message that doesn't explain the package is missing?